src/openqa_client/client.py

```diff
@@ -140,6 +143,15 @@ def do_request(self, request, retries=5, wait=10):
                 raise openqa_client.exceptions.RequestError(
                     request.method, resp.url, resp.status_code
                 )
+            if not parse:
+                return resp
+            # check if the server sent us YAML when we asked for JSON
+            contype = resp.headers.get("content-type", "")
+            if contype.startswith("text/yaml"):
+                # FullLoader should also be fine as we trust the devs,
+                # but I doubt they're gonna put anything beyond
+                # SafeLoader's capacity in the responses
```

Comment:
Yes, using SafeLoader is a good idea, (and besides that, FullLoader is still not quite safe and in development)